test/LogCorrelation.Tests/SerilogTests.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Linq;
+using Elastic.Apm;
+using Elastic.Apm.SerilogEnricher;
+using Elastic.Apm.Tests.Mocks;
+using FluentAssertions;
+using Serilog;
+using Serilog.Sinks.InMemory;
+using Xunit;
+
+namespace LogCorrelation.Tests
+{
+	public class SerilogTests
+	{
+		/// <summary>
+		/// Creates 1 simple transaction and makes sure that the log line created within the transaction has
+		/// the transaction and trace ids, and logs prior to and after the transaction do not have those.
+		/// </summary>
+		[Fact]
+		public void SerilogEnricherWithTransaction()
+		{
+			Agent.Setup(new AgentComponents(payloadSender: new NoopPayloadSender()));
```

Comment:
It might clash with other tests using agent's singleton and tests changing environment variables.
The former can be mitigated by the change proposed in https://github.com/elastic/apm-agent-dotnet/pull/636#discussion_r353559868.
The latter can be mitigated by using `TestAgentComponents` instead of `AgentComponents`.